projects/gvisor/buffer_fuzzer.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package fuzzing
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"gvisor.dev/gvisor/pkg/state"
+	"gvisor.dev/gvisor/pkg/buffer"
+)
+
+
+func doSaveAndLoad(toSave, toLoad *buffer.View) (err error) {
+	var buf bytes.Buffer
+	ctx := context.Background()
+	if _, err := state.Save(ctx, &buf, toSave); err != nil {
```

Comment:
This is neat. I'm glad the Go branch seems to work for this case, it's probably the right approach. However, this is just saving different random buffers. The code execution paths here should not really be affected by the fuzzing inputs.

We'd love any fuzzing that can be done, but I think it would need to be tweaked to provide a more meaningful result. For example:
- Fuzz the actual state data here.
- Fuzz the ELF loader in sentry/loader.
- Fuzz the network stack input path.

Note that we do system call fuzzing (and limited network fuzzzing) with syzkaller:
https://syzkaller.appspot.com/gvisor

Reply:
I have modified the fuzzer. Let me know if we can proceed with the integration.

Reply:
@amscanne Kind ping